### PR TITLE
Require a `azcore.TokenCredential` when constructing an `AzCli`

### PR DESCRIPTION
**Pull request by ellismg**
As part of this change, I adjusted `identity.GetCredentials` to panic if one was not found in the context.

### CODE REVIEW

**Comment by ellismg** [comment]
I have been working on a spike of `azd login` built on top of MSAL and was discussing with Wei what our long-term plan for how actions will denote they require authentication.  At a high level, the `AzCli` type is evolving towards feeling a lot like another SDK Client, so making it take and hold onto the credentials it will use when constructing sub-clients to do work feels correct.

File: cli/azd/cmd/sets.go [hunk 1]
```diff
@@ -96,31 +111,36 @@ var CommonSet = wire.NewSet(
 
 var InitCmdSet = wire.NewSet(
 	CommonSet,
+	newCredential,
 	newAzCliFromOptions,
```
Comment (weikanglim) [inline cli/azd/cmd/sets.go:115]: Recommend creating a new set for `AzCliSet` that includes credential + azcli

File: cli/azd/cmd/sets.go [hunk 2]
```diff
@@ -82,6 +88,15 @@ func newAzdContext() (*azdcontext.AzdContext, error) {
 	return azdCtx, nil
 }
 
+func newCredential() (azcore.TokenCredential, error) {
+	credential, err := azidentity.NewAzureCLICredential(nil)
+	if err != nil {
+		return nil, fmt.Errorf("creating credential: %w", err)
```
Comment (weikanglim) [inline cli/azd/cmd/sets.go:94]: Maybe we can opt-for a more user-friendly error message here: "failed to obtain Azure credentials: %w"

**Review by weikanglim — APPROVED** [review]
LGTM - just some minor comments. I think refactoring AzCli into its own set will help simplify usage for new commands.

File: cli/azd/cmd/sets.go (hunk 1 above)
Comment (ellismg) [reply cli/azd/cmd/sets.go]: Done.

File: cli/azd/cmd/sets.go (hunk 2 above)
Comment (ellismg) [reply cli/azd/cmd/sets.go]: Done.

**Pull request merged**